decepticonlp/extractor/basic.py

```diff
@@ -0,0 +1,43 @@
+import abc
+import random
+
+
+class ImportantWordExtractor(metaclass=abc.ABCMeta):
+    """
+        An abstract class used to represent the Keyword Extractor.SubClass should implement the extract method.
+        Methods
+        -------
+        extract(self, words: list, **kwargs)
+            extracts top_k words and returns their index.
+    """
+
+    @abc.abstractmethod
+    def extract(self, words: list, **kwargs):  # pragma: no cover
+        """Extracts Important Word and returns the Index."""
+        raise NotImplementedError
+
+    def empty_error_msg(self):
+        return "given list of words is empty or invalid"
+
+    def word_less_than_k(self):
+        return "number of words should be greater than top_k"
+
+
+class RandomImportantWordExtractor(ImportantWordExtractor):
+    """
+        A class used to apply random word extractor.
+        Methods
+        -------
+        extract(self, words: list, top_k = 1, **kwargs)
+            - extracts top_k random words and returns their index.
+        :params:
+        :words: - List of Words
+        :top_k: - Number of words to be extracted
```

Comment:
This is not the corrent syntax for docstrings, no?

Reply:
"""
        A class used to apply random word extractor.

        Methods
        -------
        extract(self, words: list, top_k = 1, **kwargs)
            - extracts top_k random words and returns their index.
        
        Parameters
        --------
        words - List of words.
        top_k - Number of words to be extracted
    """
Changed to this

Reply:
I think its ok for now